e2e/test_autoagent.sh

```
@@ -8,20 +8,26 @@ SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+export HOME="/tmp/test-autoagent-$$"
+mkdir -p "$HOME"
+# Use CLI flag --provider=mock to override .env (env var loses to .env in load order)
+MOCK_ARGS="--repl --provider=mock"
 
 echo "=== E2E: Auto-agent ==="
 
 check_binary "$BINARY"
 
 # Test 1: @mention routes to internal agent (explore uses mock provider)
-OUTPUT=$(echo -e "@explore find TODO comments\nexit" | LLAMA_PROVIDER=mock "$BINARY" --repl 2>&1) || die "@explore failed"
+OUTPUT=$(echo -e "@explore find TODO comments\nexit" | "$BINARY" $MOCK_ARGS 2>&1) || die "@explore failed"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)